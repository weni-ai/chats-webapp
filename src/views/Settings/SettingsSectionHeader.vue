<template>
  <header class="settings-view__section-header">
    <p class="section-header__title">{{ title }}</p>
    <p
      v-if="subtitle"
      class="section-header__subtitle"
    >
      {{ subtitle }}
    </p>
  </header>
</template>

<script>
export default {
  name: 'SettingsSectionHeader',

  props: {
    title: {
      type: String,
      required: true,
    },
    subtitle: {
      type: String,
      default: '',
    },
  },
};
</script>

<style lang="scss" scoped>
.settings-view__section-header {
  display: grid;
  gap: $unnnic-spacing-nano;

  color: $unnnic-color-neutral-dark;

  .section-header__title {
    font-size: $unnnic-font-size-body-lg;
    font-weight: $unnnic-font-weight-bold;
    line-height: $unnnic-font-size-body-lg + $unnnic-line-height-md;
  }
  .section-header__subtitle {
    font-size: $unnnic-font-size-body-gt;
    line-height: $unnnic-font-size-body-gt + $unnnic-line-height-md;
  }
}
</style>
