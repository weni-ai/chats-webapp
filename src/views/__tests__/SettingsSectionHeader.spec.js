import { mount } from '@vue/test-utils';
import { describe, it, expect } from 'vitest';
import SettingsSectionHeader from '@/views/Settings/SettingsSectionHeader.vue';

const createWrapper = (props) => {
  return mount(SettingsSectionHeader, {
    props,
  });
};

describe('SettingsSectionHeader.vue', () => {
  let wrapper;

  describe('Without subtitle', () => {
    beforeEach(() => {
      wrapper = createWrapper({ title: 'Section Title' });
    });

    it('should render the title correctly', () => {
      const title = wrapper.find('[data-testid=section-header-title]');

      expect(title.exists()).toBe(true);
      expect(title.text()).toBe('Section Title');
    });

    it('should not render the subtitle when it is not provided', () => {
      const subtitle = wrapper.find('[data-testid=section-header-subtitle]');

      expect(subtitle.exists()).toBe(false);
    });
  });

  describe('With subtitle', () => {
    beforeEach(() => {
      wrapper = createWrapper({
        title: 'Section Title',
        subtitle: 'Section Subtitle',
      });
    });

    it('should render the subtitle when it is provided', () => {
      const subtitle = wrapper.find('[data-testid=section-header-subtitle]');

      expect(subtitle.exists()).toBe(true);
      expect(subtitle.text()).toBe('Section Subtitle');
    });

    it('should match the snapshot', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
