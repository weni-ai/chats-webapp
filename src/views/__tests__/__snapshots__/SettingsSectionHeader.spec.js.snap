// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SettingsSectionHeader.vue > With subtitle > should match the snapshot 1`] = `
<header
  class="settings-view__section-header"
  data-v-f23b63f9=""
>
  <p
    class="section-header__title"
    data-testid="section-header-title"
    data-v-f23b63f9=""
  >
    Section Title
  </p>
  <p
    class="section-header__subtitle"
    data-testid="section-header-subtitle"
    data-v-f23b63f9=""
  >
    Section Subtitle
  </p>
</header>
`;
